### PR TITLE
[SPARK-13114][SQL] Add a test for tokens more than the fields in schema

### PR DESCRIPTION
**Pull request by HyukjinKwon**
https://issues.apache.org/jira/browse/SPARK-13114

This PR adds a test for tokens more than the fields in schema.

### CODE REVIEW

File: sql/core/src/test/resources/cars-malformed.csv [hunk 1]
```diff
@@ -0,0 +1,5 @@
+year,make,model,comment,blank
```
Comment (rxin) [inline sql/core/src/test/resources/cars-malformed.csv]: might be good if we can explain using comments what's malformed

Comment (HyukjinKwon) [inline sql/core/src/test/resources/cars-malformed.csv]: Sure.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by rxin** [comment]
Thanks - going to merge this.

**Pull request closed without merging**